renderdoc/driver/vulkan/vk_manager.h

```diff
@@ -264,18 +264,31 @@ class VulkanResourceManager : public ResourceManager<VulkanResourceManagerConfig
                      rdcarray<rdcpair<ResourceId, ImageRegionState> > &states,
                      std::map<ResourceId, ImageLayouts> &layouts);
 
+#if ENABLED(RDOC_NEW_IMAGE_STATE)
+  void RecordBarriers(std::map<ResourceId, ImageState> &states, uint32_t queueFamilyIndex,
+                      uint32_t numBarriers, const VkImageMemoryBarrier *barriers);
+
+  template <typename SerialiserType>
+  void SerialiseImageStates2(SerialiserType &ser, std::map<ResourceId, LockingImageState> &states);
+  bool DeserialiseImageRefs2(ReadSerialiser &ser, std::map<ResourceId, LockingImageState> &states);
```

Comment:
This is now called `Serialise_ImageRefs` again. The signature changed slightly from the original `Serialise_ImageRefs`, accepting a instead of an rdcarray to populate. `Serialise_ImageRefs` is also no longer templated over the serialiser.

`SerialiseImageStates2` is now just `SerialiseImageStates`.